server/src/main/java/org/elasticsearch/injection/spec/AliasSpec.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.injection.spec;
+
+/**
+ * Indicates that a type should be injected as though the requester had asked for some subtype instead.
+ */
+public record AliasSpec(Class<?> requestedType, Class<?> subtype) implements UnambiguousSpec {}
```

Comment:
Is this actually needed currently?

Reply:
Not in the PR as it currently is, but the very next milestone has a listener, which means we'll need some notion of injecting subtypes and we'll have to handle that somehow.

Rather than pull this out and then put it back in, would it make sense for me to implement `EnrichPlugin` to justify some of these features?

Reply:
On second thought, I'll just pull it out for now.